### PR TITLE
Follow up to executor pull request

### PR DESCRIPTION
**Pull request by sloretz**
Follow up to address comments #104 that didn't make it into r2b3

connects to ros2/examples#184

### CODE REVIEW

**Comment by sloretz** [comment]
Builds passed and approved, squash/merging

File: rclpy/src/rclpy/_rclpy.c [hunk 1]
```diff
@@ -121,7 +123,7 @@ rclpy_trigger_guard_condition(PyObject * Py_UNUSED(self), PyObject * args)
 {
   PyObject * pygc;
 
-  if (!PyArg_ParseTuple(args, "O", &pygc)) {
+  if (!PyArg_ParseTuple(args, "O", &pygc) || !PyCapsule_CheckExact(pygc)) {
```
Comment (mikaelarguedas) [inline rclpy/src/rclpy/_rclpy.c]: @sloretz I think this changed the behavior of the statement:
if PyArg_ParseTuple fails an error will be raised while if PyCapsule_CheckExact fails we return FALSE without raising. Is it expected or something we overlooked ?

Comment (sloretz) [reply rclpy/src/rclpy/_rclpy.c]: What was the behavior before? I think this returns false in both cases. The docs don't mention [`"O"` raising an exception, although `"O!"` does](https://docs.python.org/3/c-api/arg.html#other-objects).

Comment (mikaelarguedas) [reply rclpy/src/rclpy/_rclpy.c]: O! is only to check Object types, any failure to parse an argument will raise an exception if I read the [function description](https://docs.python.org/3/c-api/arg.html#c.PyArg_ParseTuple) correctly:
"Parse the parameters of a function that takes only positional parameters into local variables. Returns true on success; on failure, it returns false and raises the appropriate exception."

Comment (sloretz) [reply rclpy/src/rclpy/_rclpy.c]: Ah oops I missed that. Sounds like should be returning NULL on PyArg_ParseTuple failure and either raising an exception or returning False if the capsule check fails.

Comment (mikaelarguedas) [reply rclpy/src/rclpy/_rclpy.c]: I noticed that when preparing a PR to address https://github.com/ros2/rclpy/pull/120#pullrequestreview-63114812. I can fix this in the same PR. Would that conflict with your current work ?

Comment (sloretz) [reply rclpy/src/rclpy/_rclpy.c]: It would not conflict. I have a couple additions to this file, but no modifications.

**Pull request merged**